docs/tutorial/rewriter/conditional_rewrite.md

```diff
@@ -50,3 +50,56 @@ With all the necessary components in place, the pattern rewrite rule with the `m
 The final graph with the applied rewrite looks as follows:
 
 ![broadcast_rewrite](examples/img/broadcast_02.png){align=center}
+
+# Using PatternMatchContext for Advanced Condition Checking
+
+The `context` parameter passed to condition functions is an instance of {py:class}`onnxscript.rewriter.PatternMatchContext`, which provides access to additional information about the pattern match that can be useful for sophisticated condition checking.
+
+## PatternMatchContext Properties
+
+The PatternMatchContext provides the following read-only properties:
+
+- `model`: The entire ONNX model being matched
+- `graph_or_function`: The specific graph or function being matched
+- `main_root_node`: The primary root node of the matching subgraph
+- `output_values`: The output values of the matching subgraph
+- `nodes`: All nodes that are part of the matching subgraph
+
+## Example Usage
+
+Here's an example showing how to use the PatternMatchContext to implement more sophisticated condition checking:
+
+```python
+def advanced_condition_check(context, x, y, **_):
+    """Example condition function using PatternMatchContext."""
+    
+    # Access the main node of the pattern match
+    main_node = context.main_root_node
+    
+    # Check that the main_node does not have an attribute called "alpha"
+    for attr in main_node.attribute:
+        if attr.name == "alpha":
+            return False
+    
+    # Access the broader graph context and check that x occurs as a graph-input
+    model = context.model
+    input_names = [input.name for input in model.graph.input]
```

Comment:
Here, `x` is the input, not its name. So, a check like `if x in model.graph.inputs` should work, I think.

Reply:
I think the field is called `inputs` not `input` (this may be a difference between the onnx_ir we are using, and the underlying proto field name).